optparse.h

```diff
@@ -259,8 +260,12 @@ optparse(struct optparse *options, const char *optstring)
         else
             options->optarg = 0;
         return option[0];
+    default: {
```

Comment:
it's not possible for `optparse_argtype()` to return anything other than `-1 .. 2`, so this case will never execute.

Reply:
Logically yes, but I encountered one case that returned 0 from line 263. I will come back to you if I can reproduce the problem. 

By the way, isn't it so that `optparse ` should return one of follwing values?

- Option character, when an option is successfully parsed
- -1: indicate all options have been parsed 
- `'?'`: for invalid option

The `return 0` in line 263 doesn't make that much sense?